packages/components/navbar/src/NavbarSwitcher/NavbarSwitcher.tsx

```diff
@@ -41,35 +45,90 @@ function splitSpaceName(
   return [start, remainder, end];
 }
 
-type NavbarSwitcherOwnProps = CommonProps & {
-  children: ReactNode;
-  isCircle?: boolean;
-  envVariant?: 'master' | 'non-master';
-  isAlias?: boolean;
-  ariaLabel?: string;
-};
+type NavbarLoadingProps =
+  | {
+      isLoading?: true;
+      environemnt?: never;
+      space?: never;
+    }
+  | {
+      isLoading?: false;
+      environemnt?: string;
+      space?: string;
+    };
+
+type NavbarSwitcherOwnProps = CommonProps &
+  NavbarLoadingProps & {
+    isCircle?: boolean;
+    envVariant?: 'master' | 'non-master';
+    isAlias?: boolean;
+    ariaLabel?: string;
+    environment?: React.ReactNode;
+    space?: React.ReactNode;
+  };
 
 export type NavbarSwitcherProps = PropsWithHTMLElement<
   NavbarSwitcherOwnProps,
   'button'
 >;
 
+type SwitcherLabelEnvVariantProps =
+  | {
+      type: 'environment';
+      envVariant?: 'master' | 'non-master';
+    }
+  | {
+      type: 'space';
+      envVariant?: never;
+    };
+
+type SwitcherLabelProps = SwitcherLabelEnvVariantProps & {
+  value: React.ReactNode;
+  styles: ReturnType<typeof getNavbarSwitcherStyles>;
+  type: 'space' | 'environment';
+  envVariant?: 'master' | 'non-master';
+};
+
+const SwitcherLabel = ({
+  value,
+  styles,
+  envVariant,
+  type,
+}: SwitcherLabelProps) => {
+  const [start, middle, end] =
+    typeof value === 'string' ? splitString(value) : [];
+  const isEnv = type === 'environment';
+  const envColor = envVariant === 'master' ? 'green600' : 'gray600';
+
+  return start !== undefined ? (
+    <Text fontColor={isEnv ? envColor : 'gray600'}>
+      <span>{start}</span>
+      {middle && (
+        <span className={styles.switcherSpaceNameTruncation}>{middle}</span>
+      )}
+      {end && <span>{end}</span>}
+    </Text>
+  ) : (
+    <Text fontColor={isEnv ? envColor : 'gray600'}>{value}</Text>
+  );
+};
```

Comment:
Could we consolidate colors like so?

```suggestion
const SwitcherLabel = ({
  value,
  styles,
  envVariant,
  type,
}: SwitcherLabelProps) => {
  const [start, middle, end] =
    typeof value === 'string' ? splitString(value) : [];
  const isEnv = type === 'environment';
  const isMaster = envVariant === 'master';
  const fontColor = isEnv && isMaster ? 'green600' : 'gray600';
  
  return start !== undefined ? (
    <Text fontColor={fontColor}>
      <span>{start}</span>
      {middle && (
        <span className={styles.switcherSpaceNameTruncation}>{middle}</span>
      )}
      {end && <span>{end}</span>}
    </Text>
  ) : (
    <Text fontColor={fontColor}>{value}</Text>
  );
};
```

Reply:
Or rather

```typescript
switch(true) {
  case isEnv && isMaster:
    return 'green600';
  case isEnv && !isMaster || isEnv:
  default:
    return 'gray600';
}
```